### ISSUE TITLE
regression: bad error message when /usr/bin/python is not available

### ISSUE THREAD

**Issue by AceSlash** [ansible/ansible#32965]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION

```
ansible 2.4.2.0 (stable-2.4 a890108dac) last updated 2017/11/16 10:32:01 (GMT +200)
  config file = /home/user/.ansible.cfg
  configured module search path = [u'/home/user/git/external/ansible-xml']
  ansible python module location = /home/user/git/external/ansible/lib/ansible
  executable location = /home/user/git/external/ansible/bin/ansible
  python version = 2.7.14 (default, Nov  2 2017, 18:42:05) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

##### CONFIGURATION
```
ANSIBLE_PIPELINING(/home/user/.ansible.cfg) = True   
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = [u'/home/user/git/enterprise/IT/ansible/sandbox']
DEFAULT_MODULE_PATH(env: ANSIBLE_LIBRARY) = [u'/home/user/git/external/ansible-xml']
DEFAULT_VAULT_PASSWORD_FILE(/home/user/.ansible.cfg) = /home/user/.ansible_vault_pass.txt
RETRY_FILES_ENABLED(/home/user/.ansible.cfg) = False
```
##### OS / ENVIRONMENT
Host is running Fedora 26, target is running Debian 9.

##### SUMMARY
Before, when `/usr/bin/python` was not available on the target system, the error message was clear and showed `/bin/sh: 1: /usr/bin/python: not found`. Now the error is misleading:
```
TASK [Gathering Facts] ******
fatal: [sandbox-stretch]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to remote host \"10.78.5.19\". Make sure this host can be reached over ssh", "unreachable": true}
```
This is a regression.
##### STEPS TO REPRODUCE
Make sure `/usr/bin/python` is *not* installed on target machine but that the ssh connection works and then run any playbook, you will see the misleading error message.

##### EXPECTED RESULTS
The error message should clearly indicate that `/usr/bin/python` is not on target system, like it did before this regression.

##### ACTUAL RESULTS
With extra verbosity, we see the correct error message:
```
TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
Using module file /home/user/git/external/ansible/lib/ansible/modules/system/setup.py
<10.78.5.19> ESTABLISH SSH CONNECTION FOR USER: root
<10.78.5.19> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/home/user/.ansible/cp/1709e586b8 10.78.5.19 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<10.78.5.19> (127, '', '/bin/sh: 1: /usr/bin/python: not found\n')
fatal: [sandbox-stretch]: FAILED! => {
    "changed": false, 
    "module_stderr": "/bin/sh: 1: /usr/bin/python: not found\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 127
}

```
Without it, we don't see the correct error:
```
TASK [Gathering Facts] ******
fatal: [sandbox-stretch]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to remote host \"10.78.5.19\". Make sure this host can be reached over ssh", "unreachable": true}
```

**Comment by blurblah** [comment]
I've launched a Debian9 VM on AWS and tested this issue on devel branch but can't reproduce it.
`/usr/bin/python` is exist in Debian 9 image of AWS marketplace so I've tested it after moving python executable file.
- Ansible version details:
```
ansible 2.5.0 (devel 6506c4bfbc) last updated 2017/11/17 14:23:34 (GMT +900)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/Users/sunchanlee/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/sunchanlee/workspace/python/ansible/lib/ansible
  executable location = /Users/sunchanlee/workspace/python/ansible/bin/ansible
  python version = 2.7.10 (default, Jul 15 2017, 17:16:57) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```
- lsb_release of debian:
```
Distributor ID:	Debian
Description:	Debian GNU/Linux 9.2 (stretch)
Release:	9.2
Codename:	stretch
```

**Comment by AceSlash** [comment]
I'll make the test with devel latter today (cannot now), but why did you not test it with the 2.4 branch, which is the version specified in my report? It wouldn't be the first time that a bug exists only in one branch and not on devel.

**Comment by blurblah** [comment]
I think next version will be released on devel. Yes, some bugs would be there when branch out and reported issues should be checked I think. I just wanted to represent this issue is not reproduced on devel and it might be solved. That's all.

**Comment by mkrizek** [comment]
Fixed in the recent versions (`2.5+`) of Ansible.
